app/models/manageiq/providers/redhat/inventory_collection_default/infra_manager.rb

```diff
@@ -76,6 +76,39 @@ def root_folders(extra_attributes = {})
       attributes.merge!(extra_attributes)
     end
 
+    def hosts(extra_attributes = {})
+      reconnect_block = lambda do |inventory_collection, inventory_objects_index, attributes_index|
```

Comment:
we can use the same code for Vms, so we are compatible with old refresh, @agrare what else do we reconnect?

Reply:
Great idea, just hosts and vms currently.